FreeRTOS-Plus/Source/FreeRTOS-Plus-TCP/FreeRTOS_DNS.c

```diff
@@ -598,25 +841,97 @@ TickType_t uxIdentifier = 0U;
 		and return. */
 		#if( ipconfigINCLUDE_FULL_INET_ADDR == 1 )
 		{
-			ulIPAddress = FreeRTOS_inet_addr( pcHostName );
+			/* Check if the hostname given is actually an IP-addresss. */
+			#if( ipconfigUSE_IPv6 != 0 )
+			if( xFamily == FREERTOS_AF_INET6 )
+			{
+			IPv6_Address_t xAddress_IPv6;
+			BaseType_t xResult;
+
+				/* ulIPAddress does not represent an IPv4 address here. It becomes non-zero when the look-up succeeds. */
+				xResult = FreeRTOS_inet_pton6( pcHostName, xAddress_IPv6.ucBytes );
+				if( xResult == 1 )
+				{
+					ulIPAddress = 1U;
+					if( ppxAddressInfo != NULL )
+					{
+						*( ppxAddressInfo ) = pxNew_AddrInfo( pcHostName, FREERTOS_AF_INET6, xAddress_IPv6.ucBytes );
+					}
+				}
+			}
+			else
+			#endif
+			{
+				ulIPAddress = FreeRTOS_inet_addr( pcHostName );
+				if( ( ulIPAddress != 0UL) && ( ppxAddressInfo != NULL ) )
+				{
+				uint8_t *ucBytes = ( uint8_t * ) &( ulIPAddress );
+
+					*( ppxAddressInfo ) = pxNew_AddrInfo( pcHostName, FREERTOS_AF_INET4, ucBytes );
+				}
+			}
 		}
-		#endif /* ipconfigINCLUDE_FULL_INET_ADDR == 1 */
+		#endif
```

Comment:
Can we let this comment stay here? It is useful while traversing the code on a non-IDE editor.